src/chrome/client/chromeDebugAdapter/connectedCDA.ts

```diff
@@ -28,21 +29,22 @@ export class ConnectedCDA extends BaseCDAState {
         @inject(TYPES.ChromeDebugLogic) private readonly _chromeDebugAdapterLogic: ChromeDebugLogic,
         @inject(TYPES.IDomainsEnabler) private readonly _domainsEnabler: IDomainsEnabler,
         @inject(TYPES.IRuntimeStarter) private readonly _runtimeStarter: IRuntimeStarter,
-        @inject(TYPES.ISession) private readonly _session: ISession,
+        @inject(TYPES.ISession) protected readonly _session: ISession,
         @inject(TYPES.ChromeConnection) private readonly _chromeConnection: ChromeConnection,
         @inject(TYPES.TerminatingCDAProvider) private readonly _terminatingCDAProvider: TerminatingCDAProvider,
         @inject(TYPES.ChromeDebugAdapter) private readonly _chromeDebugAdapter: ChromeDebugAdapter,
         @multiInject(TYPES.IServiceComponent) private readonly _serviceComponents: IServiceComponent[],
         @inject(TYPES.BasePathTransformer) private readonly _basePathTransformer: BasePathTransformer,
-        @multiInject(TYPES.ICommandHandlerDeclarer) requestHandlerDeclarers: ICommandHandlerDeclarer[]
+        @multiInject(TYPES.ICommandHandlerDeclarer) requestHandlerDeclarers: ICommandHandlerDeclarer[],
     ) {
         super(requestHandlerDeclarers, {
             'initialize': () => { throw new Error('The debug adapter is already initialized. Calling initialize again is not supported.'); },
             'launch': () => { throw new Error("Can't launch  to a new target while connected to a previous target"); },
             'attach': () => { throw new Error("Can't attach to a new target while connected to a previous target"); },
-            'disconnect': async () => {
+            'disconnect': async (_args: DebugProtocol.DisconnectArguments) => {
+                // TODO: Add support for args.terminateDebuggee
                 this._ignoreNextDisconnectedFromWebSocket = true;
-                await this.disconnect(TerminatingReason.DisconnectedFromWebsocket);
+                await this.terminate(TerminatingReason.DisconnectedFromWebsocket);
```

Comment:
I don't think you should handle 'disconnect' as 'terminate(' because the DAP has those as different requests.

Reply:
Added disconnect method